openpype/hosts/maya/plugins/publish/extract_look.py

```diff
@@ -520,38 +628,45 @@ def _process_texture(self, filepath, do_maketx, staging, linearize, force):
-
-            if linearize:
-                self.log.info("tx: converting sRGB -> linear")
-                colorconvert = "--colorconvert sRGB linear"
-            else:
-                colorconvert = ""
-
-            config_path = get_ocio_config_path("nuke-default")
-            color_config = "--colorconfig {0}".format(config_path)
-            # Ensure folder exists
-            if not os.path.exists(os.path.dirname(converted)):
-                os.makedirs(os.path.dirname(converted))
-
-            self.log.info("Generating .tx file for %s .." % filepath)
-            maketx(
-                filepath,
-                converted,
-                # Include `source-hash` as string metadata
-                "--sattrib",
-                "sourceHash",
-                escape_space(texture_hash),
-                colorconvert,
-                color_config
-            )
-
-            return converted, COPY, texture_hash
-
-        return filepath, COPY, texture_hash
+        config_path = get_ocio_config_path("nuke-default")
+        color_config = "--colorconfig {0}".format(config_path)
+        # Ensure folder exists
+        if linearize:
+            self.log.info("tx: converting sRGB -> linear")
+            colorconvert = "--colorconvert sRGB linear"
+        else:
+            colorconvert = ""
+
+        converted = os.path.join(staging, "resources", fname + ".tx")
+        if not os.path.exists(os.path.dirname(converted)):
+            os.makedirs(os.path.dirname(converted))
+
+        if bool(processors):
+            for processor in processors:
+                if processor is MakeTX:
+                    processed_path = processor().process(filepath,
+                                                         converted,
+                                                         "--sattrib",
+                                                         "sourceHash",
+                                                         escape_space(texture_hash), # noqa
+                                                         colorconvert,
+                                                         color_config,
+                                                         )
+                    self.log.info("Generating texture file for %s .." % filepath) # noqa
+                    self.log.info(converted)
+                    if processed_path:
+                        return processed_path, COPY, texture_hash
+                    else:
+                        self.log.info("maketx has returned nothing")
+                elif processor is MakeRSTexBin:
+                    processed_path = processor().process(filepath)
+                    self.log.info("Generating texture file for %s .." % filepath) # noqa
+                    if processed_path:
+                        return processed_path, COPY, texture_hash
+                    else:
+                        self.log.info("redshift texture converter has returned nothing") # noqa
```

Comment:
Isn't it odd that we define something like a `Processor` base class but then end up needing to pass different types of input variables for the conversion - shouldn't those at least have the same method signatures so that when adding an extra type of processor it's just a means of implementing that base class?